api/src/main/java/pl/kolendateam/dadcard/race/dto/SubRaceBaseDTO.java

```diff
@@ -1,17 +1,39 @@
 package pl.kolendateam.dadcard.race.dto;
 
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
 import lombok.NoArgsConstructor;
+import pl.kolendateam.dadcard.abilitys.entity.Abilitys;
 import pl.kolendateam.dadcard.race.entity.Race;
+import pl.kolendateam.dadcard.skills.MapperSkillsToDTO;
+import pl.kolendateam.dadcard.skills.dto.RaceSkillsDTO;
+import pl.kolendateam.dadcard.skills.entity.ClassSkills;
 
 @NoArgsConstructor
 public class SubRaceBaseDTO {
     public int id;
     public String subRacesName;
     public String avatarUrl;
+    public Abilitys raceAbilitys;
+    public ArrayList <RaceSkillsDTO> raceSkills;
 
     public SubRaceBaseDTO(Race race){
         this.avatarUrl = race.getAvatarUrl();
         this.subRacesName = race.getSubRaceName();
         this.id = race.getId();
+        Gson gson = new Gson();
+        Abilitys jsonObjectAbilitys = gson.fromJson(race.getAbilitys(), Abilitys.class);
+        this.raceAbilitys = jsonObjectAbilitys;
+        Type listRaceSkill = new TypeToken<List<ClassSkills>>(){}.getType();
```

Comment:
czemu tu TypeToken<List<ClassSkills>>(){}.getType() te wąsy są tak dziwnie w środku ? z czego to wynika?

Reply:
jest to funkcja Gson zmieniająca stringa w listę obiektów Json:
List<ClassSkills>>() woła metodę parsującą listę, {} w liscie siedzą obiekty zamknięte w {}, .getType() te obiekty są jakiegoś tam typu.
Domyślam się że to tak działa.